File: src/main/java/fi/eonwe/wikilinks/WikiRedirectPage.java
package fi.eonwe.wikilinks;

import java.util.List;

/**
 */
public class WikiRedirectPage implements WikiPage {

    private final String title;
    private final long id;
    private final String target;

    public WikiRedirectPage(String title, long id, String target) {
        this.title = title;
        this.id = id;
        this.target = target;
    }

    @Override
    public boolean isRedirect() { return true; }

    @Override
    public long getId() { return id; }

    @Override
    public String getTitle() { return title; }

    public String getTarget() { return target; }

}

File: src/main/kotlin/fi/eonwe/wikilinks/fatpages/WikiRedirectPage.kt
package fi.eonwe.wikilinks.fatpages

/**
 * 
 */
data class WikiRedirectPage(val title: String, val id: Int, val target: String?) : WikiPage {
    override val isRedirect: Boolean = true
    override fun id(): Int = id

    override fun title(): String = title
}
